smarts/env/hiway_env.py

```diff
@@ -129,22 +129,62 @@ def __init__(
         if visdom:
             visdom_client = VisdomClient()
 
-        self._smarts = SMARTS(
-            agent_interfaces=agent_interfaces,
-            traffic_sim=SumoTrafficSimulation(
+        is_opendrive = HiWayEnv.check_scenario_versions(scenarios)
+        traffic_sim = None
+        if is_opendrive:
+            # We currently don't support the Native SUMO Traffic Provider or Social Agents for OpenDRIVE maps
+            if zoo_addrs:
+                warnings.warn("`zoo_addrs` should not be used with OpenDRIVE scenarios")
+                zoo_addrs = None
+            pass
+        else:
+            from smarts.core.sumo_traffic_simulation import SumoTrafficSimulation
+
+            traffic_sim = SumoTrafficSimulation(
                 headless=sumo_headless,
                 time_resolution=fixed_timestep_sec,
                 num_external_sumo_clients=num_external_sumo_clients,
                 sumo_port=sumo_port,
                 auto_start=sumo_auto_start,
                 endless_traffic=endless_traffic,
-            ),
+            )
+            zoo_addrs = zoo_addrs
+
+        self._smarts = SMARTS(
+            agent_interfaces=agent_interfaces,
+            traffic_sim=traffic_sim,
             envision=envision_client,
             visdom=visdom_client,
             fixed_timestep_sec=fixed_timestep_sec,
             zoo_addrs=zoo_addrs,
         )
 
+    @staticmethod
+    def check_scenario_versions(scenarios):
+        num_sumo = 0
+        num_opendrive = 0
+        scenario_list = Scenario.get_scenario_list(scenarios)
+        for scenario_root in scenario_list:
+            if os.path.exists(os.path.join(scenario_root, "map.net.xml")):
+                num_sumo += 1
+            elif os.path.exists(os.path.join(scenario_root, "map.xodr")):
+                num_opendrive += 1
+            else:
+                raise FileNotFoundError(
+                    f"Unable to find map in map_source={scenario_root}."
+                )
```

Comment:
Note that our dsl allows for `map_spec` to specify _any_ file name for the map source in a scenario.  The point there is to support _custom_ map types that are extensions to the `RoadMap` class, not _just_ Sumo and OpenDrive.  Users can use their own custom map builder functions.

This loop should be rewritten to reflect that.  Instead of relying on filenames, the map builder fn should be actually be used to construct the map object and then its type should be checked.  I suggest changing `_build_map()` to `build_map()` in `scenario.py` for this purpose.

Also, since the only type of map for which we currently support traffic sim is Sumo, that's the only type we really care about here.  That is, we can only support traffic if all of the maps are `SumoRoadNetworks`, but it doesn't really matter (and we don't need to count) what type they are instead.

Reply:
Done!